common/lib/dependabot/git_metadata_fetcher.rb

```diff
@@ -72,8 +72,14 @@ def fetch_upload_pack_for(uri)
       response = fetch_raw_upload_pack_for(uri)
       return response.body if response.status == 200
 
-      response_with_git = fetch_raw_upload_pack_with_git_for(uri)
-      return response_with_git.body if response_with_git.status == 200
```

Comment:
Since this is `response_with_git` rather than `response`, all the subsequent error handling is only handling the response from the initial excon request... I'm not sure if that was intentional, given the later comment `"package the command response like a HTTP response so error handling remains unchanged"` so I flipped it to `response`...

but either way could cause confusion...